.github/scripts/utils.py/update-translations.py

```diff
@@ -0,0 +1,292 @@
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def generate_build() -> int:
+    build_res = crowdin_api.translations.build_project_translation(dict({"skipUntranslatedStrings": True}))["data"]
+
+    is_finished = build_res["status"] == "finished"
+    while not is_finished:
+        time.sleep(5.0)
+        is_finished = crowdin_api.translations.check_project_build_status(build_res["id"])["data"]["status"] == "finished"
+
+    return build_res["id"]
+
+
+def download_build(build_id: int) -> CrowdinBuild:
+    download_url = crowdin_api.translations.download_project_translations(build_id)["data"]["url"]
+    build_download_req = requests.get(download_url)
+    build: CrowdinBuild = dict()
+
+    with ZipFile(BytesIO(build_download_req.content)) as build_zip:
+        file_list = build_zip.namelist()
+        file_list.sort()
+
+        for file_name in file_list:
+            if not file_name.endswith(".ini"):
+                continue
+
+            content = build_zip.read(file_name).decode()
+
+            content = content.strip()
+            if content == "":
+                continue
+
+            content = content.replace("\r\n", "\n")
+            content = content.replace("\r", "\n")
+
+            while "\n\n" in content:
+                content = content.replace("\n\n", "\n")
+
+            # Manually placed linebreaks in translations break the parsing
+            fixed_linebreaks = ""
+            previous_line = ""
+
+            for line in content.splitlines():
+                if '="' in line and not line.endswith('="'):
+                    fixed_linebreaks += "\n"
+                else:
+                    logger.warning(f"{file_name} contains a manually placed linebreak which should be fixed on Crowdin. Line: {previous_line}")
+                fixed_linebreaks += line
+
+                previous_line = line
+
+            fixed_linebreaks = fixed_linebreaks.strip()
+
+            # Parse to CrowdinBuild
+            build[file_name] = {line[:line.index("=")]: line[line.index('"') + 1:-1] for line in fixed_linebreaks.splitlines()}
+
+    return build
+
+
+def get_locale_from_path(file_path: str) -> str:
+    return file_path[file_path.rindex("/")+1:file_path.rindex(".")]
+
+
+def filter_build(build: CrowdinBuild, patterns: list[str]) -> CrowdinBuild:
+    filtered_build: CrowdinBuild = dict()
+
+    for pattern in patterns:
+        filtered_build = filtered_build | {key: build[key] for key in fnmatch.filter(list(build), pattern)}
+
+    return filtered_build
+
+
+def write_core_translations(build: CrowdinBuild) -> None:
+    for file_path, translations in build.items():
+        with open(file_path, "w") as file:
+            output = ""
+            for key, translation in translations.items():
+                output += f'{key}="{translation}"\n'
+            file.write(output)
+
+
+def write_locale_file(build: CrowdinBuild) -> None:
+    FILE_PATH = "UI/data/locale.ini"
```

Comment:
Don't use upper case naming for anything but actual constants.

_If_ these are actually constants, they should either be declared global in the file scope or  encapsulated (the latter is more elegant because it doesn't pollute the file scope, but not strictly necessary):

```Python
class OBSCrowdinConstants:
  FILE_PATH = ""
[..]

def write_locale_file(build: CrowdinBuild) -> None:
  [..]
  with open(OBSCrowdinConstants.FILE_PATH, encoding="utf-8") as file:
    [...]
```